framework/base/Model.php

```diff
@@ -239,9 +239,15 @@ public function scenarios()
      */
     public function formName()
     {
-        $reflector = new ReflectionClass($this);
+        static $formName = null;
```

Comment:
@mikehaertl There's quite a difference: class properties work on the instance level, static variables in the function scope on the class level.

If you have a class `Car` and `Car::findAll()` returns 10 different models that you call `->formName()` on, the first variant would execute the logic in `formName` 10 times (once for each instance), the latter only a single time.